src/useFieldArray.ts

```diff
@@ -33,11 +33,26 @@ const mapIds = <
 >(
   values: Partial<TFieldArrayValues>[] = [],
   keyName: TKeyName,
-): Partial<ArrayField<TFieldArrayValues, TKeyName>>[] =>
-  values.map((value: Partial<TFieldArrayValues>) => ({
-    [keyName]: generateId(),
-    ...value,
-  }));
+): Partial<ArrayField<TFieldArrayValues, TKeyName>>[] => {
+  let reservedKeyUsed = false;
+  const result = values.map((value: Partial<TFieldArrayValues>) => {
+    if (keyName in value) {
+      reservedKeyUsed = true;
+    }
+    return {
+      [keyName]: generateId(),
+      ...value,
+    };
+  });
+
+  if (process.env.NODE_ENV !== 'production' && reservedKeyUsed) {
+    console.warn(
+      `📋 useFieldArray fieldValues contain the keyName \`${keyName}\` which is reserved for use by useFieldArray. https://react-hook-form.com/api#useFieldArray`,
+    );
+  }
+
+  return result;
+};
```

Comment:
```suggestion
): Partial<ArrayField<TFieldArrayValues, TKeyName>>[] => {
  if (process.env.NODE_ENV !== 'production') {
    for (const value of values) {
      if (keyName in value) {
        console.warn(
          `📋 useFieldArray fieldValues contain the keyName \`${keyName}\` which is reserved for use by useFieldArray. https://react-hook-form.com/api#useFieldArray`,
        );

        break;
      }
    }
  }

  return values.map((value: Partial<TFieldArrayValues>) => ({
    [keyName]: generateId(),
    ...value,
  }));
};
```

Thank you for the quick PR! Love it! Just a small suggestion above.